ruby/src/IceRuby/Communicator.cpp

```diff
@@ -8,32 +8,22 @@
 #include "Logger.h"
 #include "Properties.h"
 #include "Proxy.h"
+#include "RubySliceLoader.h"
 #include "Types.h"
 #include "Util.h"
-#include "ValueFactoryManager.h"
 
 using namespace std;
 using namespace IceRuby;
 
-static VALUE _communicatorClass;
+namespace
+{
+    VALUE _communicatorClass;
 
-using CommunicatorMap = map<Ice::CommunicatorPtr, VALUE>;
-static CommunicatorMap _communicatorMap;
+    using CommunicatorMap = map<Ice::CommunicatorPtr, VALUE>;
+    CommunicatorMap _communicatorMap;
 
-extern "C" void
-IceRuby_Communicator_mark(void* p)
-{
-    auto communicator = static_cast<Ice::CommunicatorPtr*>(p);
-    try
-    {
-        auto vfm = dynamic_pointer_cast<ValueFactoryManager>((*communicator)->getValueFactoryManager());
-        assert(vfm);
-        vfm->markSelf();
-    }
-    catch (const Ice::CommunicatorDestroyedException&)
-    {
-        // Ignore. This is expected.
-    }
+    using SliceLoaderMap = map<Ice::CommunicatorPtr, Ice::SliceLoaderPtr>;
```

Comment:
Do you prefer to have the `using` even if it is only used in one place, like in this case?

Reply:
Removed.